app/src/main/org/runnerup/tracker/component/TrackerGPS.java

```diff
@@ -73,6 +74,17 @@ public ResultCode onInit(final Callback callback, Context context) {
         return ResultCode.RESULT_OK;
     }
 
+    private Integer parseAndFixInteger(int resId, String def, Context context) {
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
+        String s = preferences.getString(context.getString(resId), def);
+        if (TextUtils.isEmpty(s)) {
+            SharedPreferences.Editor prefedit = preferences.edit();
+            prefedit.putString(context.getString(resId), def);
+            prefedit.apply();
+            return Integer.parseInt(def);
```

Comment:
In this case the early return is just confusing
```suggestion
            s = def;
```